modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -401,45 +501,45 @@ impl WalletClientModule {
         })
     }
 
-    pub async fn get_deposit_address<M: Serialize + MaybeSend + MaybeSync>(
+    pub async fn allocate_deposit_address(
         &self,
-        valid_until: SystemTime,
-        extra_meta: M,
-    ) -> anyhow::Result<(OperationId, Address)> {
-        let extra_meta = serde_json::to_value(extra_meta).expect("extra meta is serializable");
+    ) -> anyhow::Result<(OperationId, Address, TweakIdx)> {
+        let now = time::now();
 
-        let (operation_id, address) = self
+        let (operation_id, address, tweak_idx) = self
             .client_ctx
             .module_autocommit(
                 |dbtx, _| {
-                    let extra_meta_inner = extra_meta.clone();
                     Box::pin(async {
-                        let (operation_id, sm, address) = self
-                            .get_deposit_address_inner(valid_until, &mut dbtx.module_dbtx())
+                        let (operation_id, address, tweak_idx) = self
+                            .get_deposit_address_inner( &mut dbtx.module_dbtx())
                             .await;
 
+                        debug!(target: LOG_CLIENT_MODULE_WALLET, %tweak_idx, %address, "Derived a new deposit address");
+
                         // Begin watching the script address
                         self.rpc
                             .watch_script_history(&address.script_pubkey())
                             .await?;
 
-                        dbtx.add_state_machines(vec![self.client_ctx.make_dyn_state(sm)])
-                            .await?;
-
-                        dbtx.add_operation_log_entry(
-                            operation_id,
-                            WalletCommonInit::KIND.as_str(),
-                            WalletOperationMeta {
-                                variant: WalletOperationMetaVariant::Deposit {
-                                    address: checked_address_to_unchecked_address(&address),
-                                    expires_at: valid_until,
+                        dbtx.module_dbtx()
```

Comment:
Why insert again here if we just inserted the new entry in `get_deposit_address_inner`? I think this is fine as-is, just curious if I'm missing something.

Reply:
Cuz I'm a distracted baboon. :disappointed_relieved: